src/trivyHelper.ts

```diff
@@ -83,28 +96,55 @@ export function getSummary(trivyStatus: number): string {
     return `- ${summary}`;
 }
 
-function getVulnerabilityIds(trivyStatus: number): string[] {
-    let vulnerabilityIds: string[] = [];
+export function printFormattedOutput() {
+    let rows = [];
+    let titles = [TITLE_VULNERABILITY_ID, TITLE_PACKAGE_NAME, TITLE_SEVERITY, TITLE_DESCRIPTION];
+    rows.push(titles);
+
+    const vulnerabilities = getVulnerabilities();
+    vulnerabilities.forEach((cve: any) => {
+        let row = [];
+        row.push(cve[KEY_VULNERABILITY_ID]);
+        row.push(cve[KEY_PACKAGE_NAME]);
+        row.push(cve[KEY_SEVERITY]);
+        row.push(cve[KEY_DESCRIPTION]);
+        rows.push(row);
+    });
+
+    let widths = [25, 20, 15, 60];
+    console.log(table.table(rows, utils.getConfigForTable(widths)));
+}
+
+function getVulnerabilityIdsBySeverity(trivyStatus: number, removeDuplicates?: boolean): any {
+    const severities: string[] = [SEVERITY_CRITICAL, SEVERITY_HIGH, SEVERITY_MEDIUM, SEVERITY_LOW, SEVERITY_UNKNOWN];
+    let vulnerabilityIdsBySeverity: any = {};
+
     if (trivyStatus == TRIVY_EXIT_CODE) {
-        const vulnerabilities = getVulnerabilities();
-        vulnerabilityIds = vulnerabilities.map(v => v[KEY_VULNERABILITY_ID]);
+        const vulnerabilities = getVulnerabilities(removeDuplicates);
+        for (let severity of severities) {
+            vulnerabilityIdsBySeverity[severity] = vulnerabilities
+                .filter(v => v[KEY_SEVERITY].toUpperCase() === severity)
+                .map(v => v[KEY_VULNERABILITY_ID]);
+        }
     }
 
-    return vulnerabilityIds;
+    return vulnerabilityIdsBySeverity;
 }
 
 function getTrivyOutput(): any {
     const path = getOutputPath();
     return fileHelper.getFileJson(path);
 }
 
-function getVulnerabilities(): any[] {
+function getVulnerabilities(removeDuplicates?: boolean): any[] {
     const trivyOutputJson = getTrivyOutput();
     let vulnerabilities: any[] = [];
     trivyOutputJson.forEach((ele: any) => {
         if (ele && ele[KEY_VULNERABILITIES]) {
             ele[KEY_VULNERABILITIES].forEach((cve: any) => {
-                vulnerabilities.push(cve);
+                if (!removeDuplicates || !vulnerabilities.some(v => v[KEY_VULNERABILITY_ID] === cve[KEY_VULNERABILITY_ID])) {
+                    vulnerabilities.push(cve);
+                }
```

Comment:
You could use this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Reply:
Not sure what the behaviour is with complex objects and whether the equality will always evaluate correctly.
Would prefer using Set in case of simple array.

Going ahead with this for now.